packages/palette/src/elements/Input/Input.tsx

```diff
@@ -45,16 +45,23 @@ export const Input: React.ForwardRefExoticComponent<
       height,
       labelOffset,
       showCounter,
+      defaultValue = "",
+      onChange,
       ...rest
     },
     ref
   ) => {
     const [boxProps, inputProps] = splitBoxProps(rest)
-    const [charCount, setCharCount] = React.useState(0)
-
-    const countChars = (e: React.ChangeEvent<HTMLInputElement>) => {
-      setCharCount(e.target.value.length)
-    }
+    const [value, setValue] = React.useState(inputProps.value || defaultValue)
```

Comment:
Bifurcating the state like this seems excessive and is going to wind up with more bugs down the line. There's a simpler fix IMO

Reply:
I guess this is what `Textarea` does, so at least there's precedent. I suspect though there's a way to externalize the notion of character limit and use it in both simultaneously.

Reply:
Would love to hear about the simpler fix :)